code/game/machinery/adv_med.dm

```diff
@@ -9,13 +9,66 @@
 	anchored = 1
 	idle_power_usage = 125
 	active_power_usage = 250
-	machine_flags = 0
-
+	var/orient = "LEFT"
+	var/scanning = 1
+	var/obj/machinery/body_scanconsole/connected = null //This will save us a lot of locates
+	machine_flags = SCREWTOGGLE | CROWDESTROY
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/fullbodyscanner,
+		/obj/item/weapon/stock_parts/scanning_module,
+		/obj/item/weapon/stock_parts/scanning_module,
+		/obj/item/weapon/stock_parts/scanning_module
+	)
 	var/mob/living/carbon/occupant
 	var/locked
 
 	l_color = "#00FF00"
 
+/obj/machinery/bodyscanner/New()
+	..()
+	RefreshParts()
+	spawn( 5 )
+		if(orient == "RIGHT")
+			update_icon()
+			generate_console(get_step(get_turf(src), WEST))
+		else
+			generate_console(get_step(get_turf(src), EAST))
+		return
+	return
+
+/obj/machinery/bodyscanner/proc/generate_console(turf/T as turf)
+	if(connected)
+		qdel(connected)
+	if(!T.density)
+		connected = new /obj/machinery/body_scanconsole(T)
+		connected.orient = src.orient
+		connected.update_icon()
+		return 1
+	else
+		return 0
+
+/obj/machinery/bodyscanner/Destroy()
+	..()
+	qdel(connected)
```

Comment:
Same with the other destroy, null out the reference. Particularly because they are hilariously calling qdel on each other on destroy.
connected.connected = null
qdel(connected)
connected = null